R/qtr.R

```diff
@@ -0,0 +1,139 @@
+#' @title Find a quarter to a date
+#'
+#' @description
+#'
+#' The qtr functions take a date input and calculate the relevant
+#' quarter-related value from it.
+#'
+#' \itemize{
+#' \item \code{qtr_year} returns the current quarter and year in either short or
+#' long format.
+#'
+#' \item \code{qtr_end} returns the last month in the quarter in either short or
+#' long format.
+#'
+#' \item \code{qtr_prev} returns the previous quarter in either short or long
+#' format.
+#' }
+#'
+#' @param date A date supplied with \code{Date} class.
+#' @param format A \code{character} string specifying the format the quarter
+#' should be displayed in. Valid options are `short` (Jan-Mar 2018) and `long`
+#' (January to March 2018).
+#'
+#' @examples
+#' x <- lubridate::dmy(c("26032012", "04052012", "23092012"))
+#'
+#' qtr_year(x, format = "long")
+#'
+#' qtr_end(x, format = "short")
+#'
+#' qtr_prev(x, format = "long")
+
+#' @export
+qtr_year <- function(date, format = c("long", "short")) {
+
+  format <- match.arg(format)
+
+  if (class(date) != "Date") {
+    stop("The current input is not a date. It must be in date format")
+  }
+
+  if (format == "long") {
+    quarter_num <- lubridate::quarter(date)
```

Comment:
`quarter_num` is calculated the same way regardless of whether the format is short or long, so I think we should move it above the `if` statement rather than calculating it here and in the `else` as well.